app/controllers/gems_adoptions_controller.rb

```diff
@@ -0,0 +1,32 @@
+class GemsAdoptionsController < ApplicationController
+  before_action :require_login
+
+  def create
+    @gems_adoption = GemsAdoption.new(gems_adoption_params)
+    @gems_adoption.user_id = current_user.id
+    if @gems_adoption.save
+      redirect_to gem_path(id: @gems_adoption.gem)
+    else
+      render :show
+    end
+  end
+
+  def destroy
+    current_user.gems_adoptions.find(params[:id]).destroy
+    redirect_to gems_path
+  end
+
+  private
+
+  def require_login
+    unless current_user.present?
+      flash[:error] =  "You must be logged in."
+      redirect_to root_url
+    end
+  end
```

Comment:
@linatodi I think this method should be moved to the `ApplicationController` so that it can be reused from any other controller that needs to check the user is logged in, what do you think?